internal/watcher/validator_test.go

```diff
@@ -76,16 +80,8 @@ func TestNewUpdateableValidator(t *testing.T) {
 			handler := file.NewHandler(afero.NewMemMapFs())
 			if tc.writeFile {
```

Comment:
```suggestion
			if tc.config != nil {
```
nit: you could remove the writeFile field and check config for nil instead